frontend/src/metabase/writeback/components/ActionCreator/SaveActionModal.tsx

```diff
@@ -0,0 +1,112 @@
+import React from "react";
+
+import _ from "underscore";
+import { t } from "ttag";
+
+import Form, { FormField, FormFooter } from "metabase/containers/FormikForm";
+import ModalContent from "metabase/components/ModalContent";
+import Modal from "metabase/components/Modal";
+
+import { CardApi } from "metabase/services";
+import validate from "metabase/lib/validate";
+
+import type Question from "metabase-lib/lib/Question";
+
+interface SaveActionModalProps {
+  question: Question;
+  onClose: () => void;
+}
+
+interface SaveActionFormData {
+  name?: string | null;
+  description?: string | null;
+  collection_id?: number | null;
+}
+
+export default function SaveActionModal({
+  question,
+  onClose,
+}: SaveActionModalProps) {
+  const initialValues: SaveActionFormData = {
+    name: question.displayName(),
+    description: question.description(),
+    collection_id: question.collectionId(),
+  };
+
+  const handleSubmit = async (questionData: SaveActionFormData) => {
+    const newQuestion = question
+      .setDisplayName(questionData.name)
+      .setDescription(questionData.description)
+      .setCollectionId(questionData.collection_id);
+
+    const response = await CardApi.create({
+      ...newQuestion.card(),
+      parameters: newQuestion.parameters(),
+      is_write: true,
+      display: "table",
+      visualization_settings: {},
+    });
+
+    if (response) {
+      setTimeout(onClose, 1000);
+      // redirect somewhere?
+    }
+  };
```

Comment:
Wonder if we should prefer moving all that to the entity layer? We have an [`Actions` entity already defined](https://github.com/metabase/metabase/blob/master/frontend/src/metabase/entities/actions.js), so we could have added a form definition to it and then use an entity form like:


```typescript
import Actions from "metabase/entities/actions";

<Actions.ModalForm
   form={Actions.forms.details}
   onSaved={...}
   onClose={...}
/>
```

We can also make the entity take care of the `is_write` flag, because it's pretty essential for actions, i.e. it won't even be an action if the flag is turned off

Reply:
Maybe we can pair on this briefly on Monday? There's a couple wrinkles with using entity forms for this particular use case:
1. We're not using the `/api/actions` endpoint to save actions, which entity forms assume, so we'll have to create a new general purpose override for this specific case
2. We need to merge form data into the card object before sending it to the api, I can't find any place we're doing that with entity forms